### DISCUSSION TITLE
Pipeline Shortcut Not Working

### DISCUSSION THREAD

**Discussion by Jeff698** [posit-dev/positron#4003] [Q&A] [answered]
Using Positron Version: 2024.07.0 (Universal) build 41 on macOS.. cmd+shift+m does not insert the pipeline (|>). 

Initially upon fresh install, it opened the Problems window. I found the keyboard short cut for this in Setting >  Keyboard Shortcuts (it was also set to cmd+shift=m) and deleted it. Now the only thing this is set to is insert pipeline. 

Now when I press cmd+shift+m... nothing happens. I've restarted positron but no change. 

Any suggestions?

**Comment by anbrav0** [comment, marked as answer]
This also happened to me when I first started playing with the first Beta release but I ended up doing the steps in this thread #3523 and it worked! 

It seems RStudios keymaps isn’t defaulted when first using Positron

**Reply by Jeff698** [reply]
Thanks! I'll take a look there!

**Comment by petetronic** [comment]
@Jeff698 Could you please share what file type you're trying to use the shortcut in? 

If it's an `r` language document in an editor that has focus, that [key binding](https://github.com/posit-dev/positron/blob/677a0d0f0a4f5f9389507c079c8b9a67699d52fd/extensions/positron-r/package.json#L342-L345) should be active as it is registered by Positron's positron-r extension by default. 

However, if you're in an r language cell in a quarto document, you'd need to satisfy the [requirements](https://github.com/posit-dev/positron/pull/3523#issuecomment-2198751543) @anbrav0 referred to in pull request #3523, i.e. Quarto installed and the setting RStudio Keymap enabled (or manually setup an [equivalent shortcut](https://github.com/posit-dev/positron/pull/3523#issuecomment-2198751543)). Note that recent builds of Positron now include the Quarto extension, but you'll still need to install the Quarto CLI yourself for now.

**Reply by Jeff698** [reply]
Thanks! I am indeed using r in a Quatro notebook. I haven't installed the Quarto CLI so that seems like a good explanation. 

Off to read #3523...

I'll report back. Appreciat the help, y'all!

**Reply by Jeff698** [reply]
That did the trick. Cmd+Shift+m now correctly adds a |> pipeline within Quarto .qmd file.

Thanks, y'all!
Jeff
